src/systemctl/systemctl-is-active.c

```diff
@@ -38,8 +38,8 @@ static int check_unit_generic(int code, const UnitActiveState good_states[], int
                 if (!arg_quiet)
                         puts(unit_active_state_to_string(active_state));
 
-                for (int i = 0; i < nb_states; ++i)
-                        if (good_states[i] == active_state) {
+                FOREACH_ARRAY(good_state, good_states, nb_states)
```

Comment:
if you touch this, maybe change nb_states type to size_t, as is appropriate for an array size